src/settings/s_decisions.php

```diff
@@ -147,7 +147,80 @@ function render(SettingValues $sv) {
         "</div>\n\n";
     Ht::stash_script("foldup(\$\$('cbfinal_open'),null,{f:\"c\"})");
+
+	//Share with Open Academic
+    echo "<h3 class=\"settings g\">Share with search engines</h3>\n";
+    echo "<p><b>Click the button to share accepted paper meta data with search engines</b></p>";
+    echo Ht::submit("share", "Share with search engines");
+
+    //wakes up API
+    $api = 'https://openacademicapi.azurewebsites.net/api/papers';
+    $opts = array('http'=> array('method' => 'GET'));
+    $context = stream_context_create($opts);
+    $post_result = file_get_contents($api, false, $context);
+   
+    if (isset($_REQUEST["share"])) {
+        $this->share($sv, $api);
+    }
+
 }
+    //sends https post request to the api
+    function share(SettingValues $sv, $api) {
+
+        $share = array();
+        $share["provider"] = "HotCRP";
+        $share["setId"] = "";
+        $share["venue"] = $sv->conf-> long_name;
+        $share["shortVenue"] = $sv->conf-> short_name;
+        $share["confSite"] = $sv->conf-> opt["conferenceSite"];
+        $share["version"] = "1.0";
+    
+        $papers = array();
+        $result = $sv->conf-> q_raw("SELECT paperID, title, authorInformation, abstract FROM Paper WHERE outcome > 0");
+        while (($row = edb_row($result))) {
+            $curr_p = [];
+            $curr_p['paperId'] = $row[0];
+            $curr_p['title'] = $row[1];
+            $curr_p['abstract'] = $row[3];
+            
+            //parse the authorInformation into an array
+            $authors = [];
+            $mul_authorRow = explode("\n", $row[2]);
+            
+            //the last element in the array is null, do not want the null
+            $mul_authorRow = array_slice($mul_authorRow, 0, count($mul_authorRow) - 1);
+            
+            foreach ($mul_authorRow as $sing_authorRow) {
+                $authorInfo = explode("\t", $sing_authorRow);
+                $curr_author = [];
+                //combine first name and last name to full name
+                $curr_author['name'] = $authorInfo[0]." ". $authorInfo[1];
+                if (strlen($authorInfo[2]) == 0) {
+                   $curr_author['srcId'] = $authorInfo[2];
+                } else {
+                   $curr_author['srcId'] = sha1($authorInfo[2]);
+                }
+                $curr_author['affiliation'] = $authorInfo[3];
+                $curr_author['order'] = 0;
+                array_push($authors, $curr_author);
+            }
+            $curr_p['authors'] = $authors;
+            array_push($papers, $curr_p);
+        }
+        $share["papers"] = $papers;
+
+        //sends HTTP POST REQUEST
+        $opts = array('http'=> array('method'=> 'POST', 'header'=> 'Content-type: application/json', 'content'=> json_encode($share)));
+        $context = stream_context_create($opts);
+        $post_result = file_get_contents($api, false, $context);
```

Comment:
Thank you for pointing this out! We definitely want to build in authentication for our web API for all of our sources including hotcrp so that we only get feeds from authenticated users. Like how arxiv.com requires user to register, the current mechanism we are thinking is that each user will have to contact us for a token that he/she can put down in an input form right above the "share" button. The token will be sent along with the feed encrypted through https and our API will verify the token. This shall be able to prevent the situation you mentioned. We would love to hear your thoughts on this!